cibyl/config.py

```diff
@@ -43,27 +44,72 @@ def _ask_user_for_overwrite():
 
 
 class Config(UserDict):
-    """Representation of a Cybil's configuration file. Even though it starts
+    """Representation of a generic configuration file. Even though it starts
     without any contents, this dictionary can be filled in with the data
     from an external yaml file. No post-processing is performed on the read
     data, as this class acts as a direct interface between the system's file
     and the app.
     """
-
-    def load(self, file):
-        """Loads the contents of a file into this object. Any contents this
-        may beforehand have are lost and replaced by the data from the file.
-        In case of error, the contents are left untouched.
-
-        :raises YAMLError: If the file could not be parsed.
+    def load(self, path=None, data=None, file=None):
+        """Loads the content of a configuration file/object and creates
+        a reference to environments."""
+        if data:
+            self.data = data
+        elif path:
+            self.data = ConfigFactory.from_path(path)
+        elif file:
+            self.data = ConfigFactory.from_file(file)
+        else:
+            self.data = ConfigFactory.from_search()
+
+
+class AppConfig(Config):
+    """Representation of a Cybil's configuration file"""
+
+    def __init__(self):
+        """AppConfig Constructor"""
+        super(Config, self).__init__()
+        self.environments = {}
+        self.plugins = {}
+
+    def load(self, path=None, data=None):
+        """Loads the content of a configuration file/object and creates
+        a reference to environments."""
+        super().load(path, data)
+        self.environments = self.data.get('environments', {})
+        self.plugins = self.data.get('plugins', [])
```

Comment:
As a tip, there is no need to have these as class attributes, you could define them as properties instead. For example:
```
@property
def environments(self):
    return self.data.get(...)
```
These are read-only and will avoid having duplicated data.